python/ray/data/_internal/planner/plan_download_op.py

```
@@ -189,8 +189,14 @@ def download_bytes_threaded(
         def load_uri_bytes(uri_path_iterator):
             """Function that takes an iterator of URI paths and yields downloaded bytes for each."""
             for uri_path in uri_path_iterator:
-                with fs.open_input_file(uri_path) as f:
-                    yield f.read()
+                try:
+                    with fs.open_input_file(uri_path) as f:
+                        yield f.read()
+                except (OSError, pa.ArrowException) as e:
```

I think we should just narrowly catch the `FileNotFoundError` to avoid masking other errors.

What's the pyarrow exception for? Can we get rid of that?

---

The OSError covers:
`FileNotFoundError`
`PermissionError`
`IsADirectoryError`
`NotADirectoryError`
`BlockingIOError`
`TimeoutError`
Those are general error we all want to handle for downloading, I would remove pa.ArrowException which is unnecessary